Docs: Flow control section (#754)
inngest/website: pages/docs/guides/throttling.mdx

export const description = 'Limit the throughput of function execution over a period of time. Ideal for working around third-party API rate limits.';

# Throttling

Throttling allows you to specify how many function runs can start within a time period. When the limit is reached, new function runs over the throttling limit will be _enqueued for the future_. Some use cases for priority include:

> * Smooth out execution of functions over time

Reviewer: ```suggestion
* Smoothing out execution of functions over time.
```

I'm not sure what that means. Could you rephrase "Smoothing out"?

> * Working around third-party API rate limits

Reviewer: ```suggestion
* Working around third-party API rate limits.
```

  }
);
```
</CodeGroup>

You can configure throttling on each function using the optional `throttle` parameter.  The options directly control the generic cell rate algorithm parameters used within the queue.

### Configuration reference

- `limit`: The total number of runs allowed to start within the given `period`.
- `period`: The period within the limit will be applied.

> - `burst`: The number of runs allow to start in the given window in a single burst. This defaults to 1, which ensure that requests are smoothed amongst the given `period`.

Reviewer: ```suggestion
- `burst`: The number of runs allowed to start in the given window in a single burst. This defaults to 1, which ensures that requests are smoothed amongst the given `period`.
```
